### PR TITLE
Adding context method to return unique id

### PR DESCRIPTION
**Pull request by sebworks**
Adding context method to return unique id

## Additions

- Adding context method to return unique id.

## Testing

- Visit  `http://localhost:8000/the-bureau/leadership-calendar/` and click on labels on the two filters. Should check/uncheck boxes on both filters.

## Review

- @kave 
- @rosskarchner 
- @anselmbradford 
- @KimberlyMunoz 

## Notes

I am using a closure but it may make more sense to use a generator or iterator.

### CODE REVIEW

File: cfgov/v1/util/util.py [hunk 1]
```diff
@@ -13,4 +13,12 @@ def id_validator(id_string, search=re.compile(r'[^a-zA-Z0-9-_]').search):
...
-    return components[0] + "".join(x.title() for x in components[1:])
\ No newline at end of file
+    return components[0] + "".join(x.title() for x in components[1:])
+
+class UniqueId(object):
+	index = 0
+
+	@classmethod
+	def get_unique_id(cls, prefix='', suffix=''):
+		cls.index = cls.index + 1
```
Comment (anselmbradford) [inline cfgov/v1/util/util.py]: Another approach is to get the current system unix timestamp and add that in place of index.

File: cfgov/jinja2/v1/_includes/macros/filters.html [hunk 2]
```diff
@@ -357,17 +357,17 @@
...
 {% macro _filter_checkbox(field, value, label) %}
     {% from 'macros/util/format/url.html' import slugify as slugify %}
-    {% set id_slug = slugify( value ) %}
+    {% set id = get_unique_id('filter_', '_') ~ slugify( value ) %}
     <input class="cf-input"
            type="checkbox"
            name="filter_{{ field }}"
            value="{{ value }}"
-           id="filter_{{ id_slug }}"
+           id="{{id}}"
```
Comment (anselmbradford) [inline cfgov/jinja2/v1/_includes/macros/filters.html]: Trivial: space out `{{ id }}` here and throughout.

**Comment by anselmbradford** [comment]
Minor comments, LGTM overall though :+1: 

Seems like helper method could be used in https://github.com/cfpb/cfgov-refresh/pull/1273#discussion_r48543903 too.

File: cfgov/v1/util/util.py (hunk 1 above)
Comment (sebworks) [inline cfgov/v1/util/util.py]: So they have `uuid.uuid4()` but I wasn't sure if that was too long. 

https://docs.python.org/2/library/uuid.html

**Comment by kave** [comment]
Can we also use this method for the FormFieldWithButton id so content no longer needs to manually input it?

**Comment by KimberlyMunoz** [comment]
> Can we also use this method for the FormFieldWithButton id so content no longer needs to manually input it?

This seems like a good idea. :bulb:

**Comment by KimberlyMunoz** [comment]
I like it. We'll have to add it into some of the existing form elements, but it'll make it much easier to make sure our forms don't have duplicate ids in the long run. :+1:

**Comment by kave** [comment]
:+1:

**Pull request merged**